sunbeam-python/sunbeam/plugins/validation/plugin.py

```diff
@@ -0,0 +1,304 @@
+            if not unit:
+                message = f"Unable to get {app} leader"
+                raise click.ClickException(message)
+
+            action_result = run_sync(
+                jhelper.run_action(
+                    unit,
+                    model,
+                    action_name,
+                    action_params or {},
+                )
+            )
+
+            if action_result.get("return-code", 0) > 1:
+                message = f"Unable to run action: {action_name}"
+                raise click.ClickException(message)
+
+            if print_stdout:
+                console.print(action_result.get("stdout").strip())
+
+    @click.command()
+    def enable_plugin(self) -> None:
+        """Enable OpenStack Integration Test Suite (tempest)."""
+        super().enable_plugin()
+
+    @click.command()
+    def disable_plugin(self) -> None:
+        """Disable OpenStack Integration Test Suite (tempest)."""
+        super().disable_plugin()
+
+    @click.command()
+    @click.option(
+        "-s",
+        "--schedule",
+        default="",
+        help=(
+            "The cron schedule expression to define when to run tempest"
+            " periodic checks. When the value is empty (default),"
+            " period checks will be disabled."
+        ),
+    )
+    def configure_validation(self, schedule: str = "") -> None:
+        """Configure validation plugin."""
+        jhelper = JujuHelper(self.client, self.snap.paths.user_data)
+        with console.status("Configuring validation plugin ..."):
+            valid, message = validate_schedule(schedule)
+            if not valid:
+                raise click.ClickException(message)
+
+            app = "tempest"
+            model = OPENSTACK_MODEL
+            unit = run_sync(jhelper.get_leader_unit(app, model))
+            if not unit:
+                message = f"Unable to get {app} leader"
+                raise click.ClickException(message)
+
+            run_sync(
+                jhelper.set_application_config(
+                    model, app, config={"schedule": schedule}
+                )
+            )
+            console.print(message)
+
+    @click.command()
+    @click.option(
+        "-s",
+        "--smoke",
+        is_flag=True,
+        default=False,
+        help="Run the smoke tests only. Equivalent to --regex=smoke.",
+    )
+    @click.option(
+        "-r",
+        "--regex",
+        default="",
+        help=(
+            "A list of regexes, whitespace separated, used to select tests from"
+            " the list."
+        ),
+    )
+    @click.option(
+        "-e",
+        "--exclude-regex",
+        default="",
+        help="A single regex to exclude tests.",
+    )
+    @click.option(
+        "-t",
+        "--serial",
+        is_flag=True,
+        default=False,
+        help="Run tests serially. By default, tests run in parallel.",
+    )
+    @click.option(
+        "--test-list",
```

Comment:
non-blocking: we can also add a shorthand option for this, e.g. `-l`

Reply:
DONE